synapseclient/client.py

```diff
@@ -349,8 +362,133 @@ def __init__(
 
         transfer_config = self._get_transfer_config()
         self.max_threads = transfer_config["max_threads"]
+        self._thread_executor = {}
+        self._process_executor = {}
+        self._md5_semaphore = {}
```

Comment:
Do we imagine the need for other semaphores?

Reply:
Perhaps around things like querying for table data later on, though I'm not certain.